ONSdigital/eq-runner-mock-cir
# Create mock CIR

### What is the context of this PR?
This PR Implements a Mock version of the CIR required by eq-runner
The Metadata model has been copied from the [repo itself](https://github.com/ONSdigital/eq-cir-fastapi/blob/main/app/models/responses.py#L21) as some of the fields are missing from the [OpenAPI spec](https://trello.com/c/AwJG3L3S/5887-create-mock-collection-instrument-registry-cir-3-sp#:~:text=to%20the%20following-,OpenAPI%20specification,-.%20This%20mock%20registry)

The following endpoints have been added:
- `/v2/retrieve_collection_instrument` to fetch a schema by uuid
- `v2/ci_metadata` to fetch a list of metadata matching the given optional parameters (return all if no params specified)

The uuids are generated by a hash of the language and schema name so that they are unique and deterministic

### How to review
- Check the requirements from the card have been implemented and that the setup of this repo matches that of the mock sds
- Make sure you can load schemas locally with the make command in the readme, and that they load correctly when running as a docker container
- Check the unit tests add suitable testing of the endpoints
